src/coreclr/jit/importer.cpp

```diff
@@ -9731,95 +9648,31 @@ GenTree* Compiler::impFixupStructReturnType(GenTree*                 op,
 
 #endif //  FEATURE_MULTIREG_RET && TARGET_ARM64
 
-    if (!compDoOldStructRetyping() && (!op->IsCall() || !op->AsCall()->TreatAsHasRetBufArg(this)))
+    if (!op->IsCall() || !op->AsCall()->TreatAsHasRetBufArg(this))
     {
         // Don't retype `struct` as a primitive type in `ret` instruction.
         return op;
     }
 
-REDO_RETURN_NODE:
-    // adjust the type away from struct to integral
-    // and no normalizing
-    if (op->gtOper == GT_LCL_VAR)
-    {
-        // It is possible that we now have a lclVar of scalar type.
-        // If so, don't transform it to GT_LCL_FLD.
-        LclVarDsc* varDsc = lvaGetDesc(op->AsLclVarCommon());
-        if (genActualType(varDsc->TypeGet()) != genActualType(info.compRetNativeType))
-        {
-            op->ChangeOper(GT_LCL_FLD);
-        }
-    }
-    else if (op->gtOper == GT_OBJ)
-    {
-        GenTree* op1 = op->AsObj()->Addr();
-
-        // We will fold away OBJ/ADDR, except for OBJ/ADDR/INDEX
```

Comment:
Your change here also removes this folding, 
This doesn't seem to be directly related to `compDoOldStructRetyping`

Reply:
It is also unreachable, the previous condition was:
```
if (!compDoOldStructRetyping() && (!op->IsCall() || !op->AsCall()->TreatAsHasRetBufArg(this)))
```
now it is 
```
if (!op->IsCall() || !op->AsCall()->TreatAsHasRetBufArg(this))
```

so op can't be anything but call.

Reply:
> > Is the SPK_EnclosingType clause in fgUpdateInlineReturnExpressionPlaceHolder also now unreachable? Would be nice if it was.
> 
> I pushed the change to delete the handling, there are no asm diffs but Jit log is smaller now.

Excellent. Thanks.